src/recon.rs

```diff
@@ -2628,11 +2628,10 @@ pub(crate) unsafe fn rav1d_recon_b_intra<BD: BitDepth>(
                             ),
                         ]);
                         let top_sb_edge_slice = if t.b.y & f.sb_step - 1 == 0 {
-                            let mut top_sb_edge: *const BD::Pixel =
-                                f.ipred_edge[0] as *mut BD::Pixel;
                             let sby = t.b.y >> f.sb_shift;
-                            top_sb_edge = top_sb_edge.offset((f.sb128w * 128 * (sby - 1)) as isize);
-                            Some(slice::from_raw_parts(top_sb_edge, f.sb128w as usize * 128))
+                            let offset = f.ipred_edge_off as isize * 0
+                                + (f.sb128w * 128 * (sby - 1)) as isize;
```

Comment:
`f.sb128w` is always positive, so I think you can make this a `usize`.

Reply:
The problem isn't `f.sb128w`, it's `sby - 1`. `sby` can sometimes be 0, which makes the entire expression negative. That's why I'm now passing the index into `rav1d_prepare_intra_edges`, so that a final non-negative offset can be calculated before indexing.

Reply:
Ohh